<launch>
	<!-- arguments -->
	<arg name="map_file" default="$(find rrt_planning)/maps/map.yaml"/>
	<arg name="world_file" default="$(find rrt_planning)/maps/map.world"/>
	<arg name="simulated" default="true" />
	
	<!-- parameters -->
	<param name="use_sim_time" value="true" if="$(arg simulated)"/>
	
	<!-- Nodes -->
	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
	
	<node name="rviz" pkg="rviz" type="rviz" respawn="true"
	args="-d $(find rrt_planning)/rviz_config/rviz.rviz" />
	
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
		<!-- basic configuration -->
		<param name="controller_frequency" value="10.0" />
		<rosparam file="$(find rrt_planning)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
		<rosparam file="$(find rrt_planning)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find rrt_planning)/config/local_costmap_params.yaml" command="load" />
		<rosparam file="$(find rrt_planning)/config/global_costmap_params.yaml" command="load" />
		<rosparam file="$(find rrt_planning)/config/dwa_local_planner_params.yaml" command="load" />

		<!-- Theta*-RRT configuration -->		
		<param name="base_global_planner" value="rrt_planning/ThetaStarRRTPlanner"/>
		<rosparam file="$(find rrt_planning)/config/theta_star.yaml" command="load" ns="ThetaStarPlanner"/>
		<rosparam file="$(find rrt_planning)/config/rrt.yaml" command="load" ns="ThetaStarRRTPlanner"/>
		<rosparam file="$(find rrt_planning)/config/differentialDrive.yaml" command="load" ns="ThetaStarRRTPlanner"/>
	</node>
	
	<include file="$(find rrt_planning)/launch/stage.launch" if="$(arg simulated)">
		<arg name="world_file" value="$(arg world_file)"/>
	</include>
	
	<include file="$(find rrt_planning)/launch/fake_localization.launch" unless="$(arg simulated)"/>
	
</launch> 
